fraction.rb

```diff
@@ -1,17 +1,61 @@
+
 class Fraction
-  def initialize(numerator, denominator)
-    if !(numerator.is_a?(Integer) && denominator.is_a?(Integer))
-      # TODO: create a proper exception for that issue
-      # TODO: handle case when one or both parameters are not Integer but could be easily transformed into it
-      # by using #to_i
-      raise 'At least some arguments are incorrect: ' + numerator.inspect + ', ' + denominator.inspect
+  def check_fraction(numerator, denominator)
+    if ! numerator.is_a?(Integer) && (denominator.is_a?(Integer) && denominator>0)
```

Comment:
It's still one cumulative validation for two numbers. [I've asked](https://github.com/nonkor/Betlab-ruby-courses/pull/1#discussion_r70066317) about splitting them and creation custom Exception classes.